dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/DSpaceObjectResolver.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.utils;
+
+import java.sql.SQLException;
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.CommunityService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.discovery.IndexableObject;
+import org.dspace.discovery.indexobject.IndexableCollection;
+import org.dspace.discovery.indexobject.IndexableCommunity;
+import org.dspace.discovery.indexobject.IndexableItem;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
```

Comment:
I think it misses java docs describing what is this for.

Reply:
I managed to get it to work and storing data in solr as expected:
```json
{
        "ip":"[MY_IP]",
        "referrer":"https://localhost/search?query=azoto",
        "dns":"[MY_DNS]",
        "userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36",
        "isBot":false,
        "id":"77efec73-84f5-4433-a039-6b7968bc3c06",
        "type":2,
        "owningColl":["9cd30711-5edb-4820-a636-c563bb05113d"],
        "owningComm":["8c4afa54-0a50-4d8f-b183-4f6a3ef827b9"],
        "time":"2023-05-04T10:30:39.381Z",
        "query":["azoto"],
        "statistics_type":"search_result",
        "rpp":0,
        "sortBy":"score",
        "sortOrder":"desc",
        "page":1,
        "uid":"43e800fb-a6a8-48a2-b245-87e3918007a9"}]
  }
```

Reply:
I've removed DSpaceObjectResolver as @tdonohue correctly pointed out it's easily replaced by `DSpaceObjectUtils.findDSpaceObject`